File: app/src/main/java/com/josegonzalez/musicsearchsample/repository/ItunesClient.java
package com.josegonzalez.musicsearchsample.repository;

import com.josegonzalez.musicsearchsample.model.TrackListDto;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ItunesClient {

    @GET("/lookup")
    Call<TrackListDto> getListOfSongsByArtistId(@Query("id") long id);

    @GET("/search?mediaType=music&")
    Call<TrackListDto> getListOfSongsWithNumberOfElements(@Query("term") String searchTerm, @Query("limit") int numberOfSongsInList);

    @GET("/search?mediaType=music&entity=song&limit=20&")
    Call<TrackListDto> getListOfSongsByAlbum(@Query("id") long id);

}

File: app/src/main/java/com/josegonzalez/musicsearchsample/repository/ItunesClient.kt
package com.josegonzalez.musicsearchsample.repository

import com.josegonzalez.musicsearchsample.model.TrackListDto
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ItunesClient {

    @GET("/lookup")
    fun getListOfSongsByArtistId(@Query("id") id: Long): Call<TrackListDto?>?

    @GET("/search?mediaType=music&")
    fun getListOfSongsWithNumberOfElements(@Query("term") searchTerm: String?, @Query("limit") numberOfSongsInList: Int): Call<TrackListDto?>?

    @GET("/search?mediaType=music&entity=song&limit=20&")
    fun getListOfSongsByAlbum(@Query("id") id: Long): Call<TrackListDto?>?

}